File: 백준/Silver/13335. 트럭/트럭.cc
#include <iostream>
#include <algorithm>
#include <cstring>
#include <queue>
#include <cmath>

using namespace std;
#define MAX 2000
#define endl '\n'

int xd[] = { -1,-1,-1,0 };
int yd[] = { -1,0,1,1 };

int N, W, L;

int main(void) {
	cin >> N >> W >> L;
	vector<int> arr(N);
	for (int i = 0;i < N;i++)
		cin >> arr[i];
	queue<int> q;
	int sum = 0, answer = 0;
	for (int i = 0;i < N;i++) {
		while (1) {
			if (q.size() == W) {
				sum -= q.front();
				q.pop();
			}
			if (sum + arr[i] <= L)
				break;
			q.push(0);
			answer++;
		}
		q.push(arr[i]);
		sum += arr[i];
		answer++;
	}
	answer += W;
	cout << answer;
}

File: 백준/Silver/13335. 트럭/트럭.py
n,w,L = map(int, input().split())
arr = list(map(int, input().split()))

q = []
sum = 0
answer = 0
for i in range(n):
    while True:
        if len(q)==w:
            sum-=q.pop(0)
        if sum+arr[i]<=L:
            break
        q.append(0)
        answer+=1
    q.append(arr[i])
    sum+=arr[i]
    answer+=1
print(answer+w)